### PR TITLE
Adding a Python Solution

### PR DESCRIPTION
**Pull request by Mohammad-Nour-Massri**

### CODE REVIEW

File: solutions/ac-DivGame.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 id: ac-DivGame
 source: AtCoder Beginner Contest
 title: Div Game
-author: Maggie Liu
+authors: Maggie Liu, Mohammad Nour Massri
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:5]:
```suggestion
author: Maggie Liu, Mohammad Nour Massri
```
needs to stay like this

File: solutions/ac-DivGame.mdx [hunk 2]
```diff
@@ -58,5 +58,30 @@ int main()
 ```
 
 </CPPSection>
+<PySection>
+
+```py
+
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:64]:
```suggestion
```

File: solutions/ac-DivGame.mdx [hunk 3]
```diff
@@ -58,5 +58,30 @@ int main()
...
+n = int(input())
+ans = 0
+#prime factorization for n
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:67]:
```suggestion
# prime factorization for n
```

File: solutions/ac-DivGame.mdx [hunk 4]
```diff
@@ -58,5 +58,30 @@ int main()
...
+for p in range(2,int(n**0.5)):
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:68]:
```suggestion
for p in range(2, int(n ** 0.5)):
```

File: solutions/ac-DivGame.mdx [hunk 5]
```diff
@@ -58,5 +58,30 @@ int main()
...
+	e = 0
+	while n % p == 0:
+		n /= p
+		e += 1
+	#here p^e divides n
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:73]:
```suggestion
	# here p^e divides n 
```
not sure if this comment is necessary

File: solutions/ac-DivGame.mdx [hunk 6]
```diff
@@ -58,5 +58,30 @@ int main()
...
+ans = 0
+#prime factorization for n
+for p in range(2,int(n**0.5)):
+	e = 0
+	while n % p == 0:
+		n /= p
+		e += 1
+	#here p^e divides n 
+	i = 1
+	#taking p^1 then p^2 and so on until e < i
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:75]:
```suggestion
	# taking p^1 then p^2 and so on until e < i
```

File: solutions/ac-DivGame.mdx [hunk 7]
```diff
@@ -58,5 +58,30 @@ int main()
...
+	while e >= i:
+		e -= i
+		ans += 1
+		i = i+1
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:79]:
```suggestion
		i += 1
```

File: solutions/ac-DivGame.mdx [hunk 8]
```diff
@@ -58,5 +58,30 @@ int main()
...
+    
+ans += n > 1 # if n>1 then it is a prime factor with e=1
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:81]:
```suggestion
ans += n > 1 # if n > 1 then it is a prime factor with e=1
```

File: solutions/ac-DivGame.mdx [hunk 9]
```diff
@@ -58,5 +58,30 @@ int main()
...
+print(ans)
+
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:83]:
```suggestion
```

File: solutions/ac-DivGame.mdx [hunk 10]
```diff
@@ -58,5 +58,30 @@ int main()
...
+```
```
Comment (ryanchou-dev) [inline solutions/ac-DivGame.mdx:85]:
```suggestion
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
nice solution

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
can you also move time complexity to be right below the `## Implementation` header?

File: solutions/ac-DivGame.mdx [hunk 11]
```diff
@@ -58,5 +58,27 @@ int main()
...
+		e += 1
+	# here p^e divides n 
+	i = 1
+	# taking p^1 then p^2 and so on until e < i
+	while e >= i:
+		e -= i
+		ans += 1
+		i += 1
+    
+ans += n > 1 # if n > 1 then it is a prime factor with e=1
```
Comment (SansPapyrus683) [inline solutions/ac-DivGame.mdx:80]:
```suggestion
ans += n > 1  # if n > 1 then it is a prime factor with e=1
```

File: solutions/ac-DivGame.mdx [hunk 12]
```diff
@@ -2,12 +2,11 @@
 id: ac-DivGame
 source: AtCoder Beginner Contest
 title: Div Game
-author: Maggie Liu
+author: Maggie Liu, Mohammad Nour Massri
 ---
 
 [Official Analysis](https://img.atcoder.jp/abc169/editorial.pdf#page=12)
```
Comment (nathangong) [inline solutions/ac-DivGame.mdx:9]:
```suggestion
```

File: solutions/ac-DivGame.mdx [hunk 13]
```diff
@@ -16,12 +15,13 @@ first choose $z = p_i^1$, then $z = p_i^2$ and so on.
 As we prime factorize and find the exponent of each prime, we can decrement the
 exponent by $1$, then $2$, and so on, as long as the exponent stays nonnegative.
 Each time we decrement the exponent, we can add $1$ to the answer.
+## Implementation
+**Time Complexity:** $\mathcal{O}(\sqrt{N})$
```
Comment (nathangong) [inline solutions/ac-DivGame.mdx:19]:
```suggestion
Each time we decrement the exponent, we can add $1$ to the answer.

## Implementation

**Time Complexity:** $\mathcal{O}(\sqrt{N})$
```

**Comment by bqi343** [comment]
@ryanchou-dev @HarryWangATX @SansPapyrus683 @nathangong

**Review by SansPapyrus683 — APPROVED** [review]
think it's acceptable i guess

File: solutions/ac-DivGame.mdx [hunk 14]
```diff
@@ -58,5 +59,27 @@ int main()
...
+		e += 1
+	# here p^e divides n 
+	i = 1
+	# taking p^1 then p^2 and so on until e < i
+	while e >= i:
+		e -= i
+		ans += 1
+		i += 1
+    
+ans += n > 1  # if n > 1 then it is a prime factor with e=1
```
Comment (nathangong) [inline solutions/ac-DivGame.mdx:81]:
```suggestion
ans += n > 1  # if n > 1 then it is a prime factor with e = 1
```

**Review by nathangong — APPROVED** [review]
Looks good, just 1 spacing change

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**